### PR TITLE
[IMP] website_slides: add populate

### PR DESCRIPTION
**Pull request by flch-odoo**
Add a populate module for eLearning to assess performance 
with lots of attendees.

### CODE REVIEW

File: addons/website_slides/populate/slide_channel_partner.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+        def _compute_next_attendee(iterator, *args):
+            partners_channel_ids = defaultdict(set)
+            for values, partner_id in zip(iterator, attendees_partner_ids):
+                partner_channel_ids = partners_channel_ids[partner_id]
+                remaining_channel_ids = list(channel_ids_set - partner_channel_ids)
+                channel_id = random.choices(
+                    remaining_channel_ids,
+                    weights=self._get_courses_weights(len(remaining_channel_ids)),
+                    k=1,
```
Comment (std-odoo) [inline addons/website_slides/populate/slide_channel_partner.py:36]: annoying that `random.choice` has no `weight` argument :/

File: addons/website_slides/populate/slide_channel_partner.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+
+class SlideChannelPartner(models.Model):
+    _inherit = 'slide.channel.partner'
+    _populate_dependencies = ['res.partner', 'slide.channel']
+    # 50% more than partners, so supports 0 to N courses per partner.
+    _populate_sizes = {'small': 150, 'medium': 3_000, 'large': 150_000}
+
+    # Will ensure some courses have many partners
+    choices_weights = {}
```
Comment (std-odoo) [inline addons/website_slides/populate/slide_channel_partner.py:17]: it's a bit annoying to have that as a class attribute, can't we use a variable inside of the method instead ?

**Review by std-odoo — COMMENTED** [review]
@flch-odoo Hi :) 

Some comments while passing, I'll have a deeper look later

Cheers

**Review by flch-odoo — COMMENTED** [review]
Thank you for this first round, @std-odoo . Prompt action, prompt reply!

File: addons/website_slides/populate/slide_channel_partner.py (hunk 2 above)
Comment (std-odoo) [reply addons/website_slides/populate/slide_channel_partner.py:17]: so you can just use lru_cache ? what do you think ?

(I don't like the small complexity it add))

**Review by std-odoo — COMMENTED** [review]
@flch-odoo Hi :) 

One small question, otherwise LGTM here :)

File: addons/website_slides/populate/slide_channel_partner.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+from odoo import models
+from odoo.tools import populate
+
+
+class SlideChannelPartner(models.Model):
+    _inherit = 'slide.channel.partner'
+    _populate_dependencies = ['res.partner', 'slide.channel']
+    # 50% more than partners, so supports 0 to N courses per partner.
+    _populate_sizes = {'small': 150, 'medium': 3_000, 'large': 150_000}
```
Comment (tde-banana-odoo) [inline addons/website_slides/populate/slide_channel_partner.py:14]: 150_000 with an underscore ?

**Review by tde-banana-odoo — APPROVED** [review]

**Comment by tde-banana-odoo** [comment]
@robodoo delegate=@flch-odoo

File: addons/website_slides/populate/slide_channel_partner.py (hunk 3 above)
Comment (flch-odoo) [reply addons/website_slides/populate/slide_channel_partner.py:14]: Yes, `150_000` is more readable than `150000`. It's [PEP 515](https://peps.python.org/pep-0515/) in Python 3.6.

Comment (tde-banana-odoo) [reply addons/website_slides/populate/slide_channel_partner.py:14]: Going to my "please don't do that" list then.

**Comment by flch-odoo** [comment]
@robodoo r+

**Pull request closed without merging**